Datastore/src/EntityPageIterator.php

```diff
@@ -61,6 +61,17 @@ public function current()
         $this->moreResultsType = isset($this->page['batch']['moreResults'])
             ? $this->page['batch']['moreResults']
             : null;
+        if ($this->config['resultLimit'] === 0 &&
+            isset($this->page['query']['limit'])
+        ) {
+            // limit is not set properly, so set the limit from parsed query
+            // which is returned by the server together with batch response
+            $limit = $this->page['query']['limit'];
+            if (isset($limit['value'])) {
+                $limit = $limit['value'];
+            }
+            $this->config['resultLimit'] = $limit;
+        }
```

Comment:
With the latest updates we discussed, I don't believe this is needed. I tested locally, commenting this out and behavior worked as expected.